templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -266,6 +266,12 @@
               <td class="label">{$form.trxn_id.label}</td>
               <td {$valueStyle}>{$form.trxn_id.html} {help id="id-trans_id"}</td>
             </tr>
+            <tr class="crm-contribution-form-block-trxn_date">
+              <td class="label">{$form.trxn_date.label}</td>
+              <td {$valueStyle}>{include file="CRM/common/jcalendar.tpl" elementName=trxn_date}<br />
```

Comment:
We should be using datepicker - note that if you define the dateType in the xml you can use addField

    $form->addField('birth_date', array('entity' => 'contact'), FALSE, FALSE);


https://github.com/litespeedmarc/civicrm-core/blob/ed0ca2486df5e2d7c40e3b8ace28b066830e3961/CRM/Contact/Form/Edit/Demographics.php#L48-L48